### PR TITLE
feat:combobox

### PR DESCRIPTION
**Pull request by karl-kallavus**
## Checklist

- [ ] You attached screenshots or added description about changes
- [ ] You use [conventional commits naming](https://www.conventionalcommits.org/en/v1.0.0/), e.g. `fix: your changes description [TicketNumber]`, `feat: your feature name [TicketNumber]`
- [ ] You have updated the documentation accordingly.

### CODE REVIEW

File: workspaces/core/src/combobox/Combobox.tsx [hunk 1]
```diff
@@ -1,180 +1,544 @@
...
+      <div
+        className={mergeClassnames('w-full relative', className)}
+      >
+        <ComboboxHeadlessUI
+          value={value as {}[]}
+          onChange={onChange}
+          disabled={disabled}
+          multiple={multiple as true}
+          nullable={nullable as true}
+          ref={ref as React.Ref<HTMLElement>}
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:77]: Can we avoid type assertions here?

File: workspaces/core/src/combobox/Combobox.tsx [hunk 2]
```diff
@@ -1,180 +1,544 @@
...
-        onChange={onChange}
+    <div
+      tabIndex={-1}
+      className={mergeClassnames(
+        'relative',
+        'flex flex-nowrap w-full align-middle items-center rounded-lg py-2 px-3 bg-gohan gap-x-2',
+        size === 'sm' && 'py-1.5 px-2 rounded',
+        (size === 'sm' || (innerLabel !== undefined && innerLabel)) && 'py-2 px-3 rounded-md gap-x-3',
+        size === 'lg' && 'py-3 px-3',
+        (size === 'xl' || (innerLabel !== undefined && innerLabel)) && 'py-3 px-4 rounded-xl gap-x-4',
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:110]: I think it makes sense to put this in a separate helper `getSize(),` to make code more clearly readable

File: workspaces/core/src/combobox/Combobox.tsx [hunk 3]
```diff
@@ -1,180 +1,544 @@
...
-        multiple={multiple}
-        nullable
+        className={mergeClassnames(
+          'flex-grow h-full border-0 !rounded-none bg-transparent px-0',
+          '!shadow-none hover:shadow-none focus:shadow-none focus-visible:shadow-none',
+          (label !== undefined && label.length > 0)
+            && (placeholder === undefined || placeholder.length === 0)
+            && 'input-xl',
+          (label !== undefined && label.length > 0)
+            && 'pt-[.75rem] input-xl-dt-label',
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:195]: please replace ` pt-[.75rem]` with `pt-3`

File: workspaces/core/src/combobox/Combobox.tsx [hunk 4]
```diff
@@ -1,180 +1,544 @@
...
+  const { popper } = useComboboxContext('Combobox.MultiSelect');
+
+  return (
+    <Listbox>
+      <Listbox.Button
+        open={open}
+        as={Trigger}
+        ref={popper?.setAnchor}
+        className={className}
+        multiple={multiple !== undefined}
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:482]: if `multiple = true,` always by default, does this check make sense? And at all, could multiple be false in this subcomponent?

File: next-docs/public/examples/combobox/Default.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+              <Combobox.Button open={open}>
+                <ControlsChevronDownSmall />
+              </Combobox.Button>
+            </Combobox.Trigger>
+            <Combobox.Transition
+              onQueryChange={setQuery0}
+            >
+              <Combobox.Options className={'min-w-[18.75rem] z-10000'}>
+                { filteredPeople0.length === 0 && query0 !== '' ? (
+                  <div className='relative cursor-default select-none py-2 px-4 text-gray-700'>
```
Comment (yarema184) [inline next-docs/public/examples/combobox/Default.tsx:65]: Let's use moon colors for examples as well

File: next-docs/public/examples/combobox/Default.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+              <Combobox.Button open={open}>
+                <ControlsChevronDownSmall />
+              </Combobox.Button>
+            </Combobox.Trigger>
+            <Combobox.Transition
+              onQueryChange={setQuery1}
+            >
+              <Combobox.Options className={'min-w-[18.75rem] z-10000'}>
+                { filteredPeople1.length === 0 && query1 !== '' ? (
+                  <div className='relative cursor-default select-none py-2 px-4 text-gray-700'>
```
Comment (yarema184) [inline next-docs/public/examples/combobox/Default.tsx:110]: Let's use moon colors for examples as well

File: workspaces/core/src/combobox/Combobox.tsx [hunk 7]
```diff
@@ -1,180 +1,544 @@
...
-        nullable
+        className={mergeClassnames(
+          'flex-grow h-full border-0 !rounded-none bg-transparent px-0',
+          '!shadow-none hover:shadow-none focus:shadow-none focus-visible:shadow-none',
+          (label !== undefined && label.length > 0)
+            && (placeholder === undefined || placeholder.length === 0)
+            && 'input-xl',
+          (label !== undefined && label.length > 0)
+            && 'pt-[.75rem] input-xl-dt-label',
+          size === undefined || size === 'md' ? 'text-base' : `text-${size}`,
```
Comment (karl-kallavus) [inline workspaces/core/src/combobox/Combobox.tsx:196]: `text-${size}`

File: workspaces/core/src/combobox/Combobox.tsx [hunk 8]
```diff
@@ -1,180 +1,544 @@
...
+          className
+        )}
+        isError={isError}
+        onFocus={() => input?.setIsFocused(true)}
+        onBlur={() => input?.setIsFocused(false)}
+        {...rest}
+      />
+      <InputInset.Label className='w-auto top-[-.125rem] !right-0 !left-0 whitespace-nowrap overflow-x-hidden'>{label}</InputInset.Label>
```
Comment (karl-kallavus) [inline workspaces/core/src/combobox/Combobox.tsx:204]: 0.5

File: workspaces/core/src/combobox/Combobox.tsx [hunk 9]
```diff
@@ -1,180 +1,544 @@
...
+  ...rest
+}: WithChildren<OptionsProps>) => {
+  const { popper } = useComboboxContext('Combobox.Options');
+  return (
+      <ComboboxHeadlessUI.Options
+        ref={popper?.setPopper}
+        style={popper?.styles?.popper}
+        {...popper?.attributes?.popper}
+        className={mergeClassnames(
+          menuWidth ? menuWidth : 'w-full max-h-[18.75rem] py-2 px-1 my-1 rounded-moon-s-md box-border bg-gohan shadow-moon-lg z-10 absolute',
```
Comment (karl-kallavus) [inline workspaces/core/src/combobox/Combobox.tsx:249]: h-72

File: next-docs/public/examples/combobox/Default.tsx [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+                displayValue={({ label }) => label}
+              />
+              <Combobox.Button open={open}>
+                <ControlsChevronDownSmall />
+              </Combobox.Button>
+            </Combobox.Trigger>
+            <Combobox.Transition
+              onQueryChange={setQuery0}
+            >
+              <Combobox.Options className={'min-w-[18.75rem] z-10000'}>
```
Comment (karl-kallavus) [inline next-docs/public/examples/combobox/Default.tsx:63]: z-50

File: next-docs/public/examples/combobox/InsetSelectStates.tsx [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+              displayValue={({ label }) => label}
+            >
+              <ControlsChevronDownSmall />
+            </Combobox.InsetSelect>
+            <Combobox.Transition
+              onQueryChange={setQuery0}
+            >
+              <Combobox.Options>
+                {filteredPeople0.length === 0 && query0 !== '' ? (
+                  <div className='relative cursor-default select-none py-2 px-4 text-gray-700'>
```
Comment (karl-kallavus) [inline next-docs/public/examples/combobox/InsetSelectStates.tsx:67]: text-gray-700 -> text-trunks

**Review by yarema184 — CHANGES_REQUESTED** [review]

File: workspaces/core/src/combobox/Combobox.tsx [hunk 12]
```diff
@@ -1,180 +1,546 @@
...
+  ...rest
+}: WithChildren<ButtonProps>) => {
+  const { size, disabled } = useComboboxContext('Combobox.Button');
+
+  return (
+    <HeadlessCombobox.Button
+      className={mergeClassnames(
+        'w-6 h-6',
+        size === 'sm' ? 'text-moon-16' : 'text-moon-24',
+        size === 'sm' && 'w-4 h-4',
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:225]: perhaps it makes sense to combine in one condition, like: 
`size === 'sm' ? 'w-4 h-4 text-moon-16' : 'text-moon-24',`

**Pull request merged**